File: app/src/main/java/com/bitflaker/lucidsourcekit/database/Converters.java
package com.bitflaker.lucidsourcekit.database;

import androidx.room.TypeConverter;

public class Converters {
    @TypeConverter
    public static boolean[] booleanArrayFromString(String value) {
        if(value == null) {
            return null;
        }
        boolean[] array = new boolean[value.length()];
        for (int i = 0; i < value.length(); i++) {
            array[i] = value.charAt(i) == '1';
        }
        return array;
    }

    @TypeConverter
    public static String booleanArrayToString(boolean[] value) {
        if(value == null){
            return null;
        }
        StringBuilder stringVal = new StringBuilder();
        for (boolean b : value) {
            stringVal.append(b ? '1' : '0');
        }
        return stringVal.toString();
    }
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/database/Converters.kt
package com.bitflaker.lucidsourcekit.database

import androidx.room.TypeConverter

class Converters {
    @TypeConverter
    fun booleanArrayFromString(value: String?): BooleanArray? {
        return value?.map { v -> v == '1' }?.toBooleanArray()
    }

    @TypeConverter
    fun booleanArrayToString(value: BooleanArray?): String? {

        if (value == null) {
            return null
        }
        val stringVal = StringBuilder()
        for (b in value) {
            stringVal.append(if (b) '1' else '0')
        }
        return stringVal.toString()
    }
}
